include/ensmallen_bits/moead/moead_impl.hpp

```diff
@@ -0,0 +1,383 @@
+        //! candidate.
+        bool flag = 0;
+        std::vector<MatType> wrapperFirst(1);
+        for (size_t idx = 0; idx < externalPopulation.size(); idx++)
+        {
+          wrapperFirst[0]=externalPopulation[idx];
+          first.clear();
+          first.resize(numObjectives, 1);
+          EvaluateObjectives(wrapperFirst, objectives, first);
+          if (Dominates(first.col(0), evaluatedCandidate.col(0)))
+          {
+            flag = 1;
+            break;
+          }
+        }
+        if (flag == 0)
+        {
+          externalPopulation.push_back(candidate[0]);
+          externalPopulationFValue.push_back(evaluatedCandidate.col(0));
+        }
+      }
+      else
+      {
+        externalPopulation.push_back(candidate[0]);
+        externalPopulationFValue.push_back(evaluatedCandidate.col(0));
+      }
+    }
+  }
+  bestFront = std::move(externalPopulation);
+
+  Callback::EndOptimization(*this, objectives, iterate, callbacks...);
+
+  ElemType performance = std::numeric_limits<ElemType>::max();
+
+  for (arma::Col<ElemType> objective: externalPopulationFValue)
+  {
+    if (arma::accu(objective) < performance)
+      performance = arma::accu(objective);
+  }
+
+  return performance;
+}
+
+//! Perform mutation of the candidate.
+template<typename MatType>
+inline void MOEAD::Mutate(MatType& child,
+    const double& rate,
+    const arma::vec& lowerBound,
+    const arma::vec& upperBound)
+{
+  size_t numVariables = lowerBound.n_elem;
+  double rnd, delta1, delta2, mutationPower, deltaq;
+  double y, yl, yu, val, xy;
+
+  for(size_t j=0; j < numVariables; j++)
+  {
+    double determiner = arma::randu();
+    if(determiner <= rate)
+    {
+      y = child[j];
+      yl = lowerBound(j);
+      yu = upperBound(j);
+      delta1 = (y-yl)/(yu-yl);
+      delta2 = (yu-y)/(yu-yl);    
+      rnd=arma::randu();
+      mutationPower=1/distributionIndex;
+      if (rnd <= 0.5)
+      {
+        xy = 1.0-delta1;
+        val = 2.0*rnd+(1.0-2.0*rnd)*(std::pow(xy,(distributionIndex+1.0)));
+        deltaq =  std::pow(val,mutationPower) - 1.0;
+      }
+      else
+      {
+        xy = 1.0-delta2;
+        val = 2.0*(1.0-rnd)+2.0*(rnd-0.5)*(pow(xy,(distributionIndex+1.0)));
+        deltaq = 1.0 - (pow(val,mutationPower));
+      }
+      y = y + deltaq*(yu-yl);
+      if (y<yl) y = yl;
+      if (y>yu) y = yu;
+      child[j] = y;
+    }
+  }
+}
+
+//! Calculate the output for single objective function using the Tchebycheff
+//! approach.
+inline double MOEAD::DecomposedSingleObjective(const arma::vec& weights,
+    const arma::vec& idealPoint,
+    const arma::vec& evaluatedCandidate)
+{
+  double value = weights(0)*std::abs(evaluatedCandidate(0)
+      - idealPoint(0));
+  for (size_t i = 1; i < numObjectives; i++)
+  {
+    value = std::min(value, weights(i)*std::abs(
+          evaluatedCandidate(i) - idealPoint(i)));
+  }
+  return value;
```

Comment:
`return arma::min(weights % arma::abs(evaluatedCandidate - idealPoint);` should do the trick here as well.